packages/compass-aggregations/src/components/pipeline-toolbar/pipeline-actions.tsx

```diff
@@ -3,26 +3,90 @@ import { connect } from 'react-redux';
 import type { ConnectedProps } from 'react-redux';
 import { Button, css, spacing } from '@mongodb-js/compass-components';
 
+import type { RootState } from '../../modules';
+import { exportToLanguage } from '../../modules/export-to-language';
+import { savingPipelineOpen } from '../../modules/saving-pipeline';
+import { saveCurrentPipeline } from '../../modules/saved-pipeline';
+import { runAggregation } from '../../modules/aggregation';
+import { changeWorkspace } from '../../modules/workspace';
+
 const pipelineActionsContainerStyles = css({});
 
 const buttonStyles = css({
   marginLeft: spacing[1],
   marginRight: spacing[1],
 });
 
-const PipelineActions: React.FunctionComponent<PipelineActionsProps> = () => {
+const PipelineActions: React.FunctionComponent<PipelineActionsProps> = ({
+  name,
+  workspace,
+  onExportToLanguage,
+  onRunAggregation,
+  onSavePipeline,
+  onChangeWorkspace,
+}) => {
   return (
     <div className={pipelineActionsContainerStyles}>
-      <Button className={buttonStyles} variant="primary">
-        Run
+      {workspace === 'builder' && (
+        <Button
+          data-testid="toolbar-run-action-button"
+          className={buttonStyles}
+          variant="primary"
+          onClick={() => {
+            onChangeWorkspace('results');
+            onRunAggregation();
```

Comment:
This mixes business logic and UI in a way that should be avoided, if changing workspace should trigger a run, this should happen from a single action, see [these](https://redux.js.org/style-guide/style-guide#avoid-dispatching-many-actions-sequentially) [two](https://redux.js.org/style-guide/style-guide#model-actions-as-events-not-setters) redux style guide recommendations